core/src/main/java/org/apache/carbondata/core/scan/model/QueryModel.java

```diff
@@ -120,7 +121,7 @@
   private boolean isFG;
 
   // whether to clear/free unsafe memory or not
-  private boolean freeUnsafeMemory = true;
+  private boolean freeUnsafeMemory;
```

Comment:
suggest to remove this variable, use system property whenever it is needed

Reply:
oh, that will need many changes.
I think it's OK to get this value from system property at the beginning and pass the value to the place where it is needed, cause the system property can be modified during processing. And in this scenario, directly using system property will cause problem.